### PR TITLE
Improve Object Download

### PR DESCRIPTION
**Pull request by dvaldivia**
Introduces native browser file download behavior 

Screen Shot 2021-07-14 at 12 54 12 PM

Test this using only the go server, since the react server won't proxy the request properly

Signed-off-by: Daniel Valdivia <18384552+dvaldivia@users.noreply.github.com>

### CODE REVIEW

File: restapi/user_objects.go [hunk 1]
```diff
@@ -68,11 +69,16 @@ func registerObjectsHandlers(api *operations.ConsoleAPI) {
 			return user_api.NewDownloadObjectDefault(int(err.Code)).WithPayload(err)
 		}
 		return middleware.ResponderFunc(func(rw http.ResponseWriter, _ runtime.Producer) {
-			x, err := io.Copy(rw, resp)
-
-			fmt.Println(x)
-			fmt.Println(err)
-
+			stat, err := resp.(*minio.Object).Stat()
```
Comment (harshavardhana) [inline restapi/user_objects.go:72]:
```suggestion
			stat, err := resp.(*minio.Object).Stat()
```

Do not ignore the error, log it or return it to the handler

File: restapi/user_objects.go [hunk 2]
```diff
@@ -68,11 +69,16 @@ func registerObjectsHandlers(api *operations.ConsoleAPI) {
...
+			// indicate it's a download to the browser, and the size of the object
+			rw.Header().Set("Content-Disposition", fmt.Sprintf("attachment; filename=\"%s\"", params.Prefix))
+			rw.Header().Set("Content-Type", "application/octet-stream")
```
Comment (harshavardhana) [inline restapi/user_objects.go:75]: stat.ContentType

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: restapi/user_objects.go (hunk 1 above)
Comment (dvaldivia) [reply restapi/user_objects.go]: solved

**Review by bexsoft — CHANGES_REQUESTED** [review]
Not sure if I have something missing, I downloaded the new code, restarted my server & frontend but when I try to download any file the page reloads and redirects me to dashboard. Not sure if some route is missing in endpoints.go:

**Comment by dvaldivia** [comment]
@bexsoft this feature doesn't work through the React development server, try this using the final go server with the commited assets on port `9090`

**Comment by bexsoft** [comment]
> @bexsoft this feature doesn't work through the React development server, try this using the final go server with the commited assets on port `9090`

I see, please add this information to the PR to avoid confusion

**Comment by dvaldivia** [comment]
@bexsoft done

**Review by bexsoft — CHANGES_REQUESTED** [review]
Please reset the state of `downloadingFiles` reducer as it keeps in loading state even when download has already completed

**Comment by dvaldivia** [comment]
@bexsoft done

**Review by bexsoft — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**